tendermint/src/types/proto/crypto.rs

```diff
@@ -1,21 +1,53 @@
-#[derive(Clone, PartialEq, Eq, ::prost::Message, serde::Serialize, serde::Deserialize)]
-pub struct PublicKey {
-    #[prost(oneof = "public_key::Sum", tags = "1, 2")]
-    pub sum: Option<public_key::Sum>,
+use tracing::error;
+
+#[derive(Clone, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
+#[serde(tag = "type", content = "value")]
+pub enum PublicKey {
+    #[serde(
+        rename = "tendermint/PubKeyEd25519",
+        with = "crate::types::serializers::bytes::base64string"
+    )]
+    Ed25519(Vec<u8>), //TODO: should we check that bytes contain a valid public key?
+    #[serde(
+        rename = "tendermint/PubKeySecp256k1",
+        with = "crate::types::serializers::bytes::base64string"
+    )]
+    Secp256k1(Vec<u8>), //TODO: should we check that bytes contain a valid public key?
+}
+
+impl PublicKey {
+    pub fn raw(&self) -> Vec<u8> {
```

Comment:
I think we can return slice of bytes instead of full copy. 

```rs 
pub fn as_bytes(&self) -> &[u8] 
```

Reply:
Agree